### PR TITLE
Move pretty printing out into Dhall.Pretty

### PR DESCRIPTION
**Pull request by ocharles**
This commit splits out the pretty printing logic from Dhall.Core into its own
module. Dhall.Core actually relies on some "internal" pretty printing functions,
so pretty printing has been moved to Dhall.Pretty.Internal - Dhall.Pretty
exposes the public facing interface.

### CODE REVIEW

**Comment by PierreR** [comment]
@ocharles out of curiosity what are the internal pretty printing functions ? If we are already using internals maybe the use of `viaShow` and `unsafeViaShow` is pertinent as these should actually be exposed and will be in a future version (https://github.com/quchen/prettyprinter/commit/1d7cb09b21d213abc16c5a7068c8c328f591dd0b)

File: src/Dhall/Pretty.hs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module Dhall.Pretty (prettyExpr) where
+
+import Dhall.Pretty.Internal
```
Comment (Gabriel439) [inline src/Dhall/Pretty.hs]: What's the reason for the separate `Dhall.Pretty.Internal` module?  Is this related to the use of mutually recursive modules?

Comment (ocharles) [reply src/Dhall/Pretty.hs]: As mentioned in the PR description:

> Dhall.Core actually relies on some "internal" pretty printing functions,
so pretty printing has been moved to Dhall.Pretty.Internal - Dhall.Pretty
exposes the public facing interface.

Does that help? It's things like `Core` needing `buildConst` (see the `Internal.hs-boot` for exactly what `Core` requires).

Comment (Gabriel439) [reply src/Dhall/Pretty.hs]: Oh, whoops!  I missed that.  That makes sense

**Comment by Gabriel439** [comment]
Thank you! :)

**Pull request merged**